changes/tasks.py

```diff
@@ -59,35 +59,43 @@ def deactivate_all_except_nurseconnect(self, change):
             {'identity': change.registrant_id, 'active': True}
         )["results"]
 
-        self.l.info("Retrieving nurseconnect messageset")
-        nc_messageset = sbm_client.get_messagesets(
-            {"short_name": "nurseconnect.hw_full.1"})["results"][0]
+        self.l.info("Retrieving nurseconnect messagesets")
+        messagesets = sbm_client.get_messagesets()["results"]
+        nc_messagesets = filter(lambda ms: 'nurseconnect' in ms['short_name'],
+                                messagesets)
+        nc_messageset_ids = [ms['id'] for ms in nc_messagesets]
 
         self.l.info("Deactivating active non-nurseconnect subscriptions")
         for active_sub in active_subs:
-            if nc_messageset["id"] != active_sub["messageset"]:
+            if active_sub["messageset"] not in nc_messageset_ids:
                 sbm_client.update_subscription(
                     active_sub["id"], {"active": False})
 
         self.l.info("Non-nurseconnect subscriptions deactivated")
         return True
 
     def deactivate_nurseconnect(self, change):
-        """ Deactivates nurseconnect subscription only
+        """ Deactivates nurseconnect subscriptions only
         """
-        self.l.info("Retrieving nurseconnect messageset")
-        nc_messageset = sbm_client.get_messagesets(
-            {"short_name": "nurseconnect.hw_full.1"})["results"][0]
+        self.l.info("Retrieving messagesets")
+        messagesets = sbm_client.get_messagesets()["results"]
+        nc_messagesets = filter(lambda ms: 'nurseconnect' in ms['short_name'],
+                                messagesets)
 
         self.l.info("Retrieving active nurseconnect subscriptions")
-        active_subs = sbm_client.get_subscriptions(
-            {'identity': change.registrant_id, 'active': True,
-             'messageset': nc_messageset["id"]}
-        )["results"]
+        active_subs = []
+        for messageset in nc_messagesets:
+            active_subs.extend(sbm_client.get_subscriptions(
+                {'identity': change.registrant_id, 'active': True,
+                 'messageset': messageset["id"]}
+            )["results"])
 
         self.l.info("Deactivating active nurseconnect subscriptions")
         for active_sub in active_subs:
             sbm_client.update_subscription(active_sub["id"], {"active": False})
+            # Record the last messageset deactivated to send to jembi
+            change.data["last_messageset"] = active_sub["messageset"]
```

Comment:
How about getting and saving the nurseconnect optout type here instead of in the `get_nurse_optout_type` function? It would save us a trip to SBM.